File: src/main/java/upbrella/be/rent/service/ConditionReportService.java
package upbrella.be.rent.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import upbrella.be.rent.dto.response.ConditionReportResponse;
import upbrella.be.rent.repository.ConditionReportRepository;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class ConditionReportService {

    private final ConditionReportRepository conditionReportRepository;

    public List<ConditionReportResponse> findAllConditionReport() {

        return conditionReportRepository.findAll()
                .stream()
                .map(ConditionReportResponse::fromConditionReport)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/upbrella/be/rent/service/ConditionReportService.kt
package upbrella.be.rent.service

import org.springframework.stereotype.Service
import upbrella.be.rent.dto.response.ConditionReportPageResponse
import upbrella.be.rent.entity.ConditionReport
import upbrella.be.rent.repository.ConditionReportRepository
import upbrella.be.rent.repository.CustomConditionReportRepository

@Service
class ConditionReportService(
    private val conditionReportRepository: ConditionReportRepository,
    private val customConditionReportRepository: CustomConditionReportRepository,
) {

    fun findAll(): ConditionReportPageResponse =
        ConditionReportPageResponse.of(customConditionReportRepository.findAllConditionReport())

    fun saveConditionReport(conditionReport: ConditionReport) {
        conditionReportRepository.save(conditionReport)
    }
}
